ql/src/java/org/apache/hadoop/hive/ql/parse/ExplainSemanticAnalyzer.java

```diff
@@ -282,4 +282,11 @@ public boolean skipAuthorization() {
     }
     return super.skipAuthorization();
   }
+
+  @Override
+  public void startAnalysis() {
+    if (conf.getBoolVar(HiveConf.ConfVars.HIVE_OPTIMIZE_HMS_QUERY_CACHE_ENABLED)) {
+      queryState.createHMSCache();
```

Comment:
why don't we need to do this for the other analyzers?
Shouldn't this be moved to some common place; like `BaseSemanticAnalyzer` or somewhere else?

Reply:
@kgyrtkirk I think the reason I initially did not have it for the other analyzers is that I could not prove it was safe easily -
I.E. For DDL I could imagine maybe there are situations in which we do not want the original metadata cached and we want to fetch the updated metadata after modification.